components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -334,6 +331,15 @@ class SitePermissionsFeature(
         return sitePermissions
     }
 
+    @Synchronized
+    internal fun getInitialSitePermissions(request: PermissionRequest): SitePermissions {
+        return if (sitePermissionsRules == null) {
+            SitePermissions(request.host, savedAt = System.currentTimeMillis())
+        } else {
+            requireNotNull(sitePermissionsRules).toSitePermissions(request.host, savedAt = System.currentTimeMillis())
```

Comment:
We could use a defensive copy here to remove the need for `requireNotNull`.  It would let the compiler handle nullability, and also (if I'm understanding your intent correctly) let us get rid of the synchronized annotation.

```kotlin
internal fun getInitialSitePermissions(request: PermissionRequest): SitePermissions {
    val sitePermissionsRules = sitePermissionsRules
    return if (sitePermissionsRules == null) {
        SitePermissions(request.host, savedAt = System.currentTimeMillis())
    } else {
        sitePermissionsRules.toSitePermissions(request.host, savedAt = System.currentTimeMillis())
    }
}
```

Reply:
Good call thanks 👍